src/lib/core/datetime/calendar-locale.ts

```diff
@@ -0,0 +1,145 @@
+   */
+  parseDate: (dateString: string) => SimpleDate;
+
+  /**
+   * Formats a SimpleDate to a string.
+   * @param date The date to format.
+   */
+  formatDate: (date: SimpleDate) => string;
+
+  /**
+   * Gets a label to display as the heading for the specified calendar month.
+   * @param date A date that falls within the month to be labeled.
+   */
+  getCalendarMonthHeaderLabel: (date: SimpleDate) => string;
+
+  /**
+   * Gets a label to display as the heading for the specified calendar year.
+   * @param date A date that falls within the year to be labeled.
+   */
+  getCalendarYearHeaderLabel: (date: SimpleDate) => string;
+}
+
+
+export class DefaultCalendarLocale implements  CalendarLocale {
+  months = SUPPORTS_INTL_API ?  this.createMonthsArray('long') :
+      [
+        'January',
+        'February',
+        'March',
+        'April',
+        'May',
+        'June',
+        'July',
+        'August',
+        'September',
+        'October',
+        'November',
+        'December'
+      ];
+
+  shortMonths = SUPPORTS_INTL_API ? this.createMonthsArray('short') :
+      ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+
+  narrowMonths = SUPPORTS_INTL_API ? this.createMonthsArray('narrow') :
+      ['J', 'F', 'M', 'A', 'M', 'J', 'J', 'A', 'S', 'O', 'N', 'D'];
+
+  days = SUPPORTS_INTL_API ? this.createDaysArray('long') :
+      ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+
+  shortDays = SUPPORTS_INTL_API ? this.createDaysArray('short') :
+      ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+
+  narrowDays = SUPPORTS_INTL_API ? this.createDaysArray('narrow') :
+      ['S', 'M', 'T', 'W', 'T', 'F', 'S'];
+
+  dates = [null].concat(SUPPORTS_INTL_API ?
+      this.createArray(31,
+          i => new Date(2017, 0, i + 1).toLocaleDateString(undefined, {day: 'numeric'})) :
+      this.createArray(31, i => String(i + 1)));
+
+  firstDayOfWeek = 0;
+
+  calendarLabel = 'Calendar';
+
+  openCalendarLabel = 'Open calendar';
+
+  parseDate(dateString: string) {
+    return SimpleDate.fromNativeDate(new Date(Date.parse(dateString)));
+  }
+
+  formatDate(date: SimpleDate) {
+    let nativeDate = date.toNativeDate();
+    return SUPPORTS_INTL_API ? nativeDate.toLocaleDateString() : nativeDate.toDateString();
+  }
+
+  getCalendarMonthHeaderLabel(date: SimpleDate) {
+    return SUPPORTS_INTL_API ?
+        date.toNativeDate().toLocaleDateString(undefined, {month: 'short', year: 'numeric'}) :
+        this.shortMonths[date.month] + ' ' + date.year;
+  }
+
+  getCalendarYearHeaderLabel(date: SimpleDate) {
+    return SUPPORTS_INTL_API ?
+        date.toNativeDate().toLocaleDateString(undefined, {year: 'numeric'}) : String(date.year);
+  }
+
+  protected createArray<T>(length: number, valueFunction: (index: number) => T): T[] {
+    return Array.apply(null, Array(length)).map((v: undefined, i: number) => valueFunction(i));
```

Comment:
Apparently that just creates an object like `{length: 7}` without adding keys for the indices, which causes `map` to not work.